Extra gay tweens

Blossomical opened the pull request:
added argument property:String (so like property can be "x", "y", "scale.y", "angle" etc... you get it so it can tween a field from a string)

USES REFLECT.SETFIELD!!!

UPDATED!

now looks like this

```lua
doTween('myTag', 'boyfriend', {x = 50, y = 800, angle = 180}, 2, 'elasticOut')

Blossomical committed:
Extra gay tweens

added argument property:String (so like property can be "x", "y", "scale.y", "angle" etc... you get it so it can tween a field from a string)

USES REFLECT.SETFIELD!!!

Blossomical committed:
put old tween stuff in Deprecated section

Unholywanderer commented:
how gay we talkin here

Blossomical committed:
Deprecated warnings

Blossomical committed:
deprecated warnings fix blah

BlueColorsin commented:
Hmmm Gay 🥴

smalnj reviewed (approved):
very gay and awesome

rodney528 reviewed (approved):
YES WE NEED THIS.

CrusherNotDrip reviewed (approved):
![image](https://user-images.githubusercontent.com/90648119/188338353-a834f9df-b68f-42ce-8b40-136d4fae2afd.png)

tposejank reviewed (approved):
very gay
very great
very good
overall 100/100
5 stars

Cherif107 commented:
hello do you want me to change this so instead of propertyToTween its just values
`doTween('myTag', 'boyfriend', {x = 50, y = 800, angle = 180}, 2, 'elasticOut')`

tposejank commented:
> hello do you want me to change this so instead of propertyToTween its just values `doTween('myTag', 'boyfriend', {x = 50, y = 800, angle = 180}, 2, 'elasticOut')`

hello cherry yes
just make as usable as possible

DMMaster636 commented on source/FunkinLua.hx:
```diff
@@ -2706,6 +2616,136 @@ class FunkinLua {
 			FlxG.sound.music.fadeOut(duration, toValue);
 			luaTrace('musicFadeOut is deprecated! Use soundFadeOut instead.', false, true);
 		});
+		Lua_helper.add_callback(lua, "doTweenX", function(tag:String, vars:String, value:Dynamic, duration:Float, ease:String) {
```
i've realized that it'd be hard to make it in arrays... so yeah, oopsies

Cherif107 commented:
ok i actually realized that luabitch cant support more than 5 arguments

Blossomical committed:
lua jit is a bitch so i changed the args :sob:

since luabitch doesnt support 5+ arguments the only way is to replace tag and variable to an array [tag, variable]

Cherif107 commented:
yeah now it has to look like this :sob:
![image](https://user-images.githubusercontent.com/78418245/188465225-14a27ce2-38f1-4ca0-b413-fc0b5c6b992d.png)

ImaginationSuperHero528 commented on source/FunkinLua.hx:
```diff
@@ -1088,68 +1088,18 @@ class FunkinLua {
-		});
-		Lua_helper.add_callback(lua, "doTweenAlpha", function(tag:String, vars:String, value:Dynamic, duration:Float, ease:String) {
-			var penisExam:Dynamic = tweenShit(tag, vars);
+		Lua_helper.add_callback(lua, "doTween", function(tagNVars:Array<String>, property:String, value:Dynamic, duration:Float, ease:String) {
```
I'm confused... how would I use this?

kingyomoma replied:
```lua
doTween({'tag', 'iconP1'}, 'x', 10, 2.2, 'linear')
```

tweens iconP1.x to 10 over a duration of 10 seconds with a `linear` easing type

Cherif107 commented:
hi guys so it'll look like this from now on
`doTween('myTag', 'boyfriend', {x = 50, y = 800, angle = 180}, 2, 'elasticOut')`

ImaginationSuperHero528 commented:
> hi guys so it'll look like this from now on `doTween('myTag', 'boyfriend', {x = 50, y = 800, angle = 180}, 2, 'elasticOut')`

What about noteTween? Same right?

Blossomical committed:
woooo

ImaginationSuperHero528 commented on source/FunkinLua.hx:
```diff
@@ -1171,60 +1123,20 @@ class FunkinLua {
-			cancelTween(tag);
+		
+		// bisexual note tween
+		Lua_helper.add_callback(lua, "noteTween", function(tagNnote:Array<Dynamic>, property:String, value:Dynamic, duration:Float, ease:String) {
```
Ok, maybe not the same.

Blossomical committed:
i frogor the note twens

Blossomical committed:
minorish spelling mistake

Blossomical committed:
smelling mistake #2

Blossomical committed:
anotehr spelling mistake

rodney528 reviewed (approved):
Can't wait for the merge (at least I hope it happens)!

xLinkmain commented:
YOOOOOO swag 😎😎😎

Blossomical committed:
fix

ImaginationSuperHero528 commented:
@ShadowMario plz fucking aprove this shit!

MaybeMaru reviewed (approved):
very gay indeed

rodney528 reviewed (approved):
I'm honestly surprised this hasn't been merged.

Luisinhi010 commented:
> I'm honestly surprised this hasn't been merged.

same

jayrattt reviewed (approved):
amazing and gay

tylerandari12 commented:
I was wondering why things werent like this from the start...
Also yes gay tween...

CoolingTool commented:
very fruity

tylerandari12 commented:
I might as well learn now...

How do I approve changes?

NotNimu commented:
Ultra gay

ImaginationSuperHero528 commented:
How would scale.x work?

CoolingTool commented:
like doTween('myTag', 'boyfriend.scale', {x = 2}, 2, 'elasticOut') i assume

elchango69 commented:
get out of here snicker

On Fri, Apr 28, 2023 at 11:24 AM Cherry ***@***.***> wrote:

> Closed #10433 <https://github.com/ShadowMario/FNF-PsychEngine/pull/10433>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ShadowMario/FNF-PsychEngine/pull/10433#event-9126962508>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7OIQJ3MGJVS6FU4D55Q6SDXDP4NNANCNFSM6AAAAAAQEPLG3E>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>

-- 
**Disclaimer: This message and any attachments are intended for the use of 
the addresses(s) only and may be confidential and/or legally privileged. If 
the reader is not the intended recipient, DO NOT READ, notify sender and 
delete this message. In addition, be aware that any disclosure, copying, 
distribution or use of the contents of this message is strictly prohibited. 
The contents of this message, while possibly falling under the exceptions 
of the Inspection of Public Records ACT [NMSA Chapter 14, Article2] may be 
subject to inspection by the public.
